subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/JavaPluginHelper.java

```diff
@@ -32,25 +35,42 @@
  * This class exists to avoid adding these methods to the {@code JavaPlugin} itself,
  * and thus avoids adding these methods to the public API.
  */
-public class JavaPluginHelper {
+public final class JavaPluginHelper {
 
     private JavaPluginHelper() {
         // Private to prevent instantiation.
     }
 
     /**
-     * Gets the main Java component. This method assumes the Java plugin is applied.
+     * Gets the Java component. This method assumes the Java plugin is applied.
      *
      * @throws GradleException If the {@code java} component does not exist.
      */
-    public static JvmSoftwareComponentInternal getJavaComponent(Project project) {
+    public static JvmSoftwareComponent getJavaComponent(Project project) {
         SoftwareComponent component = project.getComponents().findByName(JvmConstants.JAVA_COMPONENT_NAME);
 
-        if (!(component instanceof JvmSoftwareComponentInternal)) {
+        if (!(component instanceof JvmSoftwareComponent)) {
```

Comment:
❕ There would be an edge case here where a script does **not** apply the java plugin, yet creates a component with the same name and type.  Should this if check if the plugin has been applied as well?

Reply:
Good question. In my mind the goal would be the opposite. The `java-library`, `application`, etc plugins should be very simple and should only really need to create a component and add a feature to it. The eventual goal would be to package up most of the wiring in the components and features and not need the `java` plugin as well. 

If anything, we should update the wording here to not call out the Java plugin explicitly